sdk/storage/Azure.Storage.Queues/src/QueueMessageCodec.cs

```diff
@@ -19,7 +19,14 @@ public static string EncodeMessageBody(BinaryData binaryData, QueueMessageEncodi
                 case QueueMessageEncoding.UTF8:
                     return binaryData.ToString();
                 case QueueMessageEncoding.Base64:
-                    return Convert.ToBase64String(binaryData.ToArray());
+                    if (MemoryMarshal.TryGetArray(binaryData.ToMemory(), out var segment))
+                    {
+                        return Convert.ToBase64String(segment.Array);
```

Comment:
You will need to use the overload that takes an offset and length in case the BinaryData was created from an ArraySegment.

Reply:
done.